<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 7 Session info | scAnnotatiONT paper companion</title>
  <meta name="description" content=" 7 Session info | scAnnotatiONT paper companion" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content=" 7 Session info | scAnnotatiONT paper companion" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="https://github.com/LehmannN/scAnnotatiONT" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 7 Session info | scAnnotatiONT paper companion" />
  
  
  

<meta name="author" content="Nathalie Lehmann" />


<meta name="date" content="2021-08-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="a-tool-and-pipeline-to-improve-annotation-for-scrna-seq-analyses.html"/>

<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-3.5.1/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.18/datatables.js"></script>
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<script src="libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="libs/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><i class="fa fa-check"></i><b>1</b> Impact of discrepant reference annotations on scRNA-seq analyses</a>
<ul>
<li class="chapter" data-level="1.1" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#discrepancies-in-reference-annotations"><i class="fa fa-check"></i><b>1.1</b> Discrepancies in reference annotations</a>
<ul>
<li class="chapter" data-level="1.1.1" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#define-paths-and-get-reference-annotations"><i class="fa fa-check"></i><b>1.1.1</b> Define paths and get reference annotations</a></li>
<li class="chapter" data-level="1.1.2" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#basic-stats-on-gtf-files"><i class="fa fa-check"></i><b>1.1.2</b> Basic stats on GTF files</a></li>
<li class="chapter" data-level="1.1.3" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#run-nextflow-pipeline"><i class="fa fa-check"></i><b>1.1.3</b> Run nextflow pipeline</a></li>
<li class="chapter" data-level="1.1.4" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#mikado"><i class="fa fa-check"></i><b>1.1.4</b> Mikado</a></li>
<li class="chapter" data-level="1.1.5" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#gene-lengths"><i class="fa fa-check"></i><b>1.1.5</b> Gene lengths</a></li>
<li class="chapter" data-level="1.1.6" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#utr-lengths"><i class="fa fa-check"></i><b>1.1.6</b> 3â€™UTR lengths</a></li>
<li class="chapter" data-level="1.1.7" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#venn-diagrams"><i class="fa fa-check"></i><b>1.1.7</b> Venn Diagrams</a></li>
<li class="chapter" data-level="1.1.8" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#gffcompare"><i class="fa fa-check"></i><b>1.1.8</b> GffCompare</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html"><a href="impact-of-discrepant-reference-annotations-on-scrna-seq-analyses.html#impact-of-reference-choice-on-scrna-seq-analyses"><i class="fa fa-check"></i><b>1.2</b> Impact of reference choice on scRNA-seq analyses</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="incomplete-annotations-induce-signal-loss.html"><a href="incomplete-annotations-induce-signal-loss.html"><i class="fa fa-check"></i><b>2</b> Incomplete annotations induce signal loss</a></li>
<li class="chapter" data-level="3" data-path="approaches-to-improve-transcriptome-with-long-reads.html"><a href="approaches-to-improve-transcriptome-with-long-reads.html"><i class="fa fa-check"></i><b>3</b> Approaches to improve transcriptome with long reads</a></li>
<li class="chapter" data-level="4" data-path="impact-of-re-annotation-on-scrna-seq-analyses.html"><a href="impact-of-re-annotation-on-scrna-seq-analyses.html"><i class="fa fa-check"></i><b>4</b> Impact of re-annotation on scRNA-seq analyses</a></li>
<li class="chapter" data-level="5" data-path="validation-of-novel-genes-with-scrna.html"><a href="validation-of-novel-genes-with-scrna.html"><i class="fa fa-check"></i><b>5</b> Validation of novel genes with scRNA</a></li>
<li class="chapter" data-level="6" data-path="a-tool-and-pipeline-to-improve-annotation-for-scrna-seq-analyses.html"><a href="a-tool-and-pipeline-to-improve-annotation-for-scrna-seq-analyses.html"><i class="fa fa-check"></i><b>6</b> A tool and pipeline to improve annotation for scRNA-seq analyses</a></li>
<li class="chapter" data-level="7" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>7</b> Session info</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scAnnotatiONT paper companion</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="session-info" class="section level1" number="7">
<h1><span class="header-section-number"> 7</span> Session info</h1>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="session-info.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-conda-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.4 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /import/bc_workspaces/csb/lehmann/conda/envs/scRNA/lib/libopenblasp-r0.3.12.so
## 
## locale:
##  [1] LC_CTYPE=fr_FR.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=fr_FR.UTF-8        LC_COLLATE=fr_FR.UTF-8    
##  [5] LC_MONETARY=fr_FR.UTF-8    LC_MESSAGES=fr_FR.UTF-8   
##  [7] LC_PAPER=fr_FR.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=fr_FR.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ggpubr_0.4.0       gridExtra_2.3      ggplot2_3.3.3      RColorBrewer_1.1-2
## [5] eulerr_6.1.0       dplyr_1.0.5        bookdown_0.22      knitr_1.33        
## 
## loaded via a namespace (and not attached):
##  [1] matrixStats_0.58.0          bitops_1.0-6               
##  [3] GenomeInfoDb_1.26.7         tools_4.0.3                
##  [5] backports_1.2.1             bslib_0.2.4                
##  [7] utf8_1.2.1                  R6_2.5.0                   
##  [9] DT_0.18                     DBI_1.1.1                  
## [11] BiocGenerics_0.36.0         colorspace_2.0-0           
## [13] withr_2.4.1                 tidyselect_1.1.0           
## [15] curl_4.3                    compiler_4.0.3             
## [17] Biobase_2.50.0              DelayedArray_0.16.3        
## [19] labeling_0.4.2              rtracklayer_1.50.0         
## [21] sass_0.3.1                  scales_1.1.1               
## [23] stringr_1.4.0               digest_0.6.27              
## [25] Rsamtools_2.6.0             foreign_0.8-81             
## [27] rmarkdown_2.7               rio_0.5.26                 
## [29] XVector_0.30.0              pkgconfig_2.0.3            
## [31] htmltools_0.5.1.1           MatrixGenerics_1.2.1       
## [33] highr_0.8                   fastmap_1.1.0              
## [35] htmlwidgets_1.5.3           rlang_0.4.10               
## [37] readxl_1.3.1                rstudioapi_0.13            
## [39] shiny_1.6.0                 farver_2.1.0               
## [41] jquerylib_0.1.3             generics_0.1.0             
## [43] jsonlite_1.7.2              crosstalk_1.1.1            
## [45] BiocParallel_1.24.1         zip_2.1.1                  
## [47] car_3.0-10                  RCurl_1.98-1.3             
## [49] magrittr_2.0.1              GenomeInfoDbData_1.2.4     
## [51] Matrix_1.3-2                Rcpp_1.0.6                 
## [53] munsell_0.5.0               S4Vectors_0.28.1           
## [55] fansi_0.4.2                 abind_1.4-5                
## [57] lifecycle_1.0.0             stringi_1.5.3              
## [59] yaml_2.2.1                  carData_3.0-4              
## [61] SummarizedExperiment_1.20.0 zlibbioc_1.36.0            
## [63] grid_4.0.3                  parallel_4.0.3             
## [65] promises_1.2.0.1            forcats_0.5.1              
## [67] crayon_1.4.1                lattice_0.20-41            
## [69] Biostrings_2.58.0           haven_2.4.0                
## [71] hms_1.0.0                   polylabelr_0.2.0           
## [73] pillar_1.6.0                GenomicRanges_1.42.0       
## [75] ggsignif_0.6.1              stats4_4.0.3               
## [77] XML_3.99-0.6                glue_1.4.2                 
## [79] evaluate_0.14               data.table_1.14.0          
## [81] vctrs_0.3.7                 httpuv_1.5.5               
## [83] cellranger_1.1.0            polyclip_1.10-0            
## [85] gtable_0.3.0                purrr_0.3.4                
## [87] tidyr_1.1.3                 assertthat_0.2.1           
## [89] xfun_0.22                   openxlsx_4.2.3             
## [91] mime_0.10                   xtable_1.8-4               
## [93] broom_0.7.6                 rstatix_0.7.0              
## [95] later_1.1.0.1               tibble_3.1.0               
## [97] GenomicAlignments_1.26.0    IRanges_2.24.1             
## [99] ellipsis_0.3.1</code></pre>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="a-tool-and-pipeline-to-improve-annotation-for-scrna-seq-analyses.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
